### PR TITLE
[SPARK-15280] [Input/Output] Refactored OrcOutputWriter and moved serialization to a new class.

### PR DESCRIPTION
**Pull request by seyfe**
## What changes were proposed in this pull request?

Refactoring: Separated ORC serialization logic from OrcOutputWriter and moved to a new class called OrcSerializer.
## How was this patch tested?

Manual tests & existing tests.

### CODE REVIEW

**Comment by seyfe** [comment]
Hi @liancheng and @yhuai,

Would you mind looking at this pull request? I would appreciate that.

Thanks,
Ergin

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala [hunk 1]
```diff
@@ -167,39 +166,69 @@ private[sql] class DefaultSource
   }
 }
 
-private[orc] class OrcOutputWriter(
-    path: String,
-    bucketId: Option[Int],
-    dataSchema: StructType,
-    context: TaskAttemptContext)
-  extends OutputWriter with HiveInspectors {
+class OrcSerializer(dataSchema: StructType, conf: Configuration) extends HiveInspectors {
```
Comment (yhuai) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala]: Can we add `private[orc]` modifier?

Comment (seyfe) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala]: Sure, I added that. Thanks.

**Comment by yhuai** [comment]
test this please

**Comment by yhuai** [comment]
@seyfe Can you also update the description (since we are doing a refactoring instead of adding a new public class)?

**Comment by yhuai** [comment]
LGTM. I will merge this to master and branch 2.0 once the description is updated. Also regarding "How was this patch tested?", we can say that manual tests and existing tests.

**Comment by seyfe** [comment]
Thanks @yhuai. I updated the description accordingly.

**Comment by yhuai** [comment]
Merging to master and branch 2.0.

**Pull request closed without merging**